homeassistant/components/meteo_france/sensor.py

```diff
@@ -1,168 +1,262 @@
-    SENSOR_TYPE_ICON,
-    SENSOR_TYPE_NAME,
-    SENSOR_TYPE_UNIT,
+    ENTITY_API_DATA_PATH,
+    ENTITY_CLASS,
+    ENTITY_ENABLE,
+    ENTITY_ICON,
+    ENTITY_NAME,
+    ENTITY_UNIT,
     SENSOR_TYPES,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
-STATE_ATTR_FORECAST = "1h rain forecast"
-STATE_ATTR_BULLETIN_TIME = "Bulletin date"
-
 
 async def async_setup_entry(
     hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
 ) -> None:
     """Set up the Meteo-France sensor platform."""
-    city = entry.data[CONF_CITY]
-    client = hass.data[DOMAIN][city]
-    weather_alert_client = hass.data[DOMAIN]["weather_alert_client"]
-
-    alert_watcher = None
-    datas = client.get_data()
-    # Check if a department code is available for this city.
-    if "dept" in datas:
-        try:
-            # If yes create the watcher DepartmentWeatherAlert object.
-            alert_watcher = await hass.async_add_executor_job(
-                DepartmentWeatherAlert, datas["dept"], weather_alert_client
-            )
-            _LOGGER.info(
-                "Weather alert watcher added for %s in department %s",
-                city,
-                datas["dept"],
-            )
-        except ValueError as exp:
-            _LOGGER.error(
-                "Unexpected error when creating the weather alert sensor for %s in department %s: %s",
-                city,
-                datas["dept"],
-                exp,
+    coordinator_forecast = hass.data[DOMAIN][entry.entry_id][COORDINATOR_FORECAST]
+    coordinator_rain = hass.data[DOMAIN][entry.entry_id][COORDINATOR_RAIN]
+    coordinator_alert = hass.data[DOMAIN][entry.entry_id][COORDINATOR_ALERT]
+
+    entities = []
+    for sensor_type in SENSOR_TYPES:
+        if sensor_type == "next_rain":
+            if coordinator_rain:
+                entities.append(MeteoFranceRainSensor(sensor_type, coordinator_rain))
+                _LOGGER.debug(
+                    "Next rain sensor added for %s.",
+                    coordinator_forecast.data.position["name"],
+                )
+
+        elif sensor_type == "weather_alert":
+            if coordinator_alert:
+                entities.append(MeteoFranceAlertSensor(sensor_type, coordinator_alert))
+                _LOGGER.debug(
+                    "Weather alert sensor for department n°%s added with %s.",
+                    coordinator_forecast.data.position["dept"],
+                    coordinator_forecast.data.position["name"],
+                )
+        elif sensor_type in ["rain_chance", "freeze_chance", "snow_chance"]:
+            if coordinator_forecast.data.probability_forecast:
+                entities.append(MeteoFranceSensor(sensor_type, coordinator_forecast))
+                _LOGGER.debug(
+                    "Sensor %s added for %s.",
+                    sensor_type,
+                    coordinator_forecast.data.position["name"],
+                )
+            else:
+                _LOGGER.info(
```

Comment:
warning

Reply:
Same comment. This Log inform that some infos are missing because meteo France don't provide the same level of details for all cities (generaly for not in France cities)

Reply:
```suggestion
                _LOGGER.warning(
```